src/routes/preparation.js

```diff
@@ -0,0 +1,349 @@
+ * @body userId int
+ * @body volumeMl int
+ * @body nicotineMillipercent int
+ * @body PreparationsDiluents array
+ */
+router.post(
+  '/',
+  authenticate(),
+  [
+    body('recipeId')
+      .isNumeric()
+      .toInt(),
+    body('userId')
+      .isNumeric()
+      .toInt(),
+    body('volumeMl')
+      .isNumeric()
+      .toInt(),
+    body('nicotineMillipercent').isString(),
+    body('PreparationsDiluents')
+      .isArray()
+      .withMessage('array')
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+    const { recipeId, userId, volumeMl, nicotineMillipercent } = req.body;
+
+    log.info(`request for NEW PREPARATION`);
+    try {
+      // Check recipe exists
+      const recipeCheck = await Recipe.findOne({
+        where: {
+          id: req.body.recipeId
+        }
+      });
+
+      if (recipeCheck.length === 0) {
+        // Recipe doesn't exist
+        res.statusMessage = `Recipe ID ${req.body.recipeId} Requested for Preparation Does Not Exist`;
+        return res.status(204).end();
+      }
+      // Create the preparation, with associations
+      const result = await Preparation.create(
+        {
+          recipeId,
+          userId,
+          volumeMl,
+          nicotineMillipercent,
+          viewCount: 0,
+          PreparationsDiluents: req.body.PreparationsDiluents // Array of diluents
+        },
+        {
+          include: [
+            {
+              model: PreparationsDiluents,
+              required: true
+            }
+          ]
+        }
+      );
+
+      if (result.length === 0) {
+        return res.status(204).end();
+      }
+
+      res.type('application/json');
+      res.json(result);
+    } catch (error) {
+      log.error(error.message);
+      res.status(500).send(error.message);
+    }
+  }
+);
+/**
+ * PUT Update a Preparation
+ * - Doesn't allow the Recipe ID or User ID to change
+ * @param id int
+ * @body volumeMl int
+ * @body nicotineMillipercent int
+ * @body PreparationsDiluents array
+ */
+router.put(
+  '/:id',
+  authenticate(),
+  [
+    param('id')
+      .isNumeric()
+      .toInt(),
+    body('volumeMl')
+      .isNumeric()
+      .toInt(),
+    body('nicotineMillipercent').isString(),
+    body('PreparationsDiluents')
+      .isArray()
+      .withMessage('array')
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+    const { id } = req.params;
+
+    log.info(`update preparation id ${id}`);
+    try {
+      // Check Preparation exists
+      const preparationCheck = await Preparation.findOne({
+        where: {
+          id
+        }
+      });
+
+      if (preparationCheck.length === 0) {
+        // Preparation doesn't exist
+        res.statusMessage = `Preparation ID ${req.params.id} Does Not Exist`;
```

Comment:
I dunno how I feel about setting a custom response to the 204, it's going to be wonky on the front-end to try and parse that out of the response for _some_ 204s but not all of them, I think an HTTP 400 with that text as a JSON response is a better way.